### PR TITLE
Update Docs to show QueryBuilder::setParameters with array instead of ArrayCollection

### PR DESCRIPTION
**Pull request by pableu**
See https://github.com/doctrine/orm/pull/11292#issuecomment-1960135295.

### CODE REVIEW

File: docs/en/reference/query-builder.rst [hunk 1]
```diff
@@ -284,10 +284,10 @@ following syntax:
     // $qb instanceof QueryBuilder
 
     // Query here...
-    $qb->setParameters(new ArrayCollection([
-        new Parameter('1', 'value for ?1'),
-        new Parameter('2', 'value for ?2')
-    ]));
+    $qb->setParameters([
+        '1' => 'value for ?1',
+        '2' => 'value for ?2',
```
Comment (pableu) [inline docs/en/reference/query-builder.rst]: I don't know, do you think the array-keys should be strings (`'1'`, `'2'`) or ints ( `1`, `2`)?

Comment (greg0ire) [reply docs/en/reference/query-builder.rst]: PHP will convert numeric keys to ints when used as array keys: https://3v4l.org/dajmE

So I think you might as well use integers directly.

Comment (pableu) [reply docs/en/reference/query-builder.rst]: :+1:

**Comment by greg0ire** [comment]
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/2.18.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

**Comment by pableu** [comment]
Squashed :-)

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Review by derrabus — CHANGES_REQUESTED** [review]
With this change, we would promote the usage of a deprecated API. I don't think that this is what we want.

**Comment by greg0ire** [comment]
Indeed, see the original discussion

**Pull request closed without merging**